### PR TITLE
Handle security realm saying user is not found

### PR DESCRIPTION
**Pull request by halkeye**
# Description

See [JENKINS-53188](https://issues.jenkins-ci.org/browse/JENKINS-53188).

tl;dr - Users can exist, but deleted in security realm (example, LDAP user is deleted, but still exists in jenkins). Add exception handling to not let things break.

Was able to test this manually by deleting the user from the disk, but leaving it in users.xml

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by halkeye** [comment]
Jenkins ci timed out uploading artifacts

**Comment by imeredith** [comment]
LGTM +9000 for test

**Comment by sophistifunk** [comment]
bbm-bueno
